### DISCUSSION TITLE
CGNS 4.4.0 Released

### DISCUSSION THREAD

**Discussion by brtnfld** [CGNS/CGNS#717] [CGNStalk]
<h2>INTRODUCTION</h2>
<p>This document describes the difference between CGNS 4.3.0 and<br>
CGNS 4.4.0 and contains information on known problems in<br>
CGNS 4.4.0.</p>
<p>Links to the CGNS current released source code can be found at:</p>
<p><a href="http://cgns.org/download.html" rel="nofollow">http://cgns.org/download.html</a></p>
<p>User documentation for the current release can be found at:</p>
<p><a href="http://cgns.org/CGNS_docs_current/midlevel/index.html" rel="nofollow">http://cgns.org/CGNS_docs_current/midlevel/index.html</a></p>
<p>For more information, see the CGNS home page:</p>
<p><a href="http://cgns.org" rel="nofollow">http://cgns.org</a></p>
<h2>CONTENTS</h2>
<ul>
<li>New Features</li>
<li>Bug Fixes since CGNS 4.3.0</li>
<li>Known Problems</li>
</ul>
<h1>New Features</h1>

Configuration:
-------------
N/A

Library:
--------
N/A

Parallel Library:
-----------------
Added new parallel APIs  (C and Fortran: cgp_coord_multi_read_data, cgp_coord_multi_write_data, cgp_field_multi_read_data,  cgp_field_multi_write_data, cgp_array_multi_write_data and cgp_array_multi_read_data) which use HDF5's multi-dataset read and write APIs. The new  HDF5 feature (version 1.14.0) allows access to multiple datasets with a single I/O call. The use of these new CGNS routines can improve performance when data is accessed across several datasets from all processes since the data can be aggregated in the HDF5 or MPI-I/O library.

Fortran Library:
----------------
N/A

Tools:
------
N/A

<h1>Bug Fixes since CGNS 4.3.0 release</h1>

Library:
-------
- Removed possible overflow in cgi_error, #343 #701 
- Increased maximum open file limit default
- Added control for HDF5 tuning routine "H5Pset_elink_file_cache_size,"  which is useful when many external links are used, #351
- Fixed backward compatibility issues #702:
   (a) Older CGNS files may not have enabled creation ordering in HDF5, causing issues with the number of  Bases returned.
   (b) Fixed H5Fset_libver_bounds for 1.8 HDF5 when trying to read a CGNS file in 1.10 HDF5 file format and added a check which opens, for modifying, a CGNS file that uses HDF5 1.10 format. It will maintain 1.10 format capability since downgrading the file format to HDF5 1.8 requires repacking it (see h5repack from HDF5).

Fortran:
--------
- Removed "NULL" and "UserDefined" Fortran parameters #349 

  There is a "NULL" function in the Fortran Standard so it could conflict with this definition. Therefore, CG_NULL should be used instead. Older codes need to update to use CG_NULL instead.
  Removed the "UserDefined" parameter. CG_UserDefined should be used instead. Older codes need to update to use CG_UserDefined.

Configuration:
-------------

Tools:
------

</code></pre></div>
<h1>Known Problems</h1>

************ FORTRAN ************

* A gfortran bug in version 10.2 broke Fortran mapping and caused cg_goto_f 
 to segfault. All other versions of gfortran are suitable.
 (ref. CGNS-246, GNU BUG 100149)

* A bug in gfortran (all versions) causes cg_configure_f to fail,
  GNU BUG 99982. Other Fortran compilers are OK.

************ FORTRAN END ********

************ CGNSVIEW ************

* cgnsview for OSX is not viewing properly, and cgnsview under Windows 
  may fail to compile due to tcl/tk incompatibility. 

************ CGNSVIEW END ********

* For other issues, see https://github.com/CGNS/CGNS/issues
</code></pre></div>
<h1>Supported Platforms</h1>
<p>The following platforms are supported and have been tested for this release.<br>
They are built with autotools unless specified otherwise.</p>
<div class="snippet-clipboard-content position-relative overflow-auto"><pre><code>Linux 3.10.0-1127.10.1.el7    gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)
#1 SMP ppc64 GNU/Linux        g++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)
(echidna)                     GNU Fortran (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)

Linux 2.6.32-573.18.1.el6     IBM XL C/C++ V13.1
#1 SMP ppc64 GNU/Linux        IBM XL Fortran V15.1
(ostrich)                     

Linux 3.10.0-327.10.1.el7     GNU C (gcc), Fortran (gfortran), C++ (g++)
#1 SMP x86_64 GNU/Linux       compilers:
(jelly/kituo/moohan)          Version 4.8.5 20150623 (Red Hat 4.8.5-39)
                                 Versions 4.9.3, 5.3.0, 6.3.0, 7.2.0
                                          8.3.0, 9.1.0
                              Intel(R) C (icc), C++ (icpc), Fortran (icc)
                              compilers:
                                 Version 17.0.0.098 Build 20160721
                              MPICH 3.3 compiled with GCC 7.2.0
                              OpenMPI 4.0.0 compiled with GCC 7.2.0
                              NAG Fortran Compiler Release 7.0(Yurakuchho) Build 7011

SunOS 5.11 11.4.5.12.5.0      Sun C 5.15 SunOS_sparc 2017/05/30
32- and 64-bit                Studio 12.6 Fortran 95 8.8 SunOS_sparc 2017/05/30
(hedgehog)                    Sun C++ 5.15 SunOS_sparc 2017/05/30

Windows 10 x64                Visual Studio 2015 w/ Intel Fortran 18 (CMake)
                              Visual Studio 2017 w/ Intel Fortran 19 (CMake)
                              Visual Studio 2019 w/ Intel Fortran 19 (CMake)
                              Visual Studio 2019 w/ MSMPI 10.1 (CMake)

macOS High Sierra 10.13.6     Apple LLVM version 10.0.0 (clang-1000.10.44.4)
64-bit                        gfortran GNU Fortran (GCC) 6.3.0
(bear)                        Intel icc/icpc/ifort version 19.0.4.233 20190416

macOS Mojave 10.14.6          Apple LLVM version 10.0.1 (clang-1001.0.46.4)
64-bit                        gfortran GNU Fortran (GCC) 6.3.0
(bobcat)                      Intel icc/icpc/ifort version 19.0.4.233 20190416
</code></pre></div>
<h1>Tested Configuration Features Summary</h1>
<div class="snippet-clipboard-content position-relative overflow-auto"><pre><code>In the table below
      y   = tested
      n   = not tested in this release
      x   = not working in this release
</code></pre></div>

Platform | C | C[1] | Fortran | Fortran [1]
-- | -- | -- | -- | --
SunOS 5.11 32-bit | y | n | y | n
SunOS 5.11 64-bit | y | n | y | n
Windows 10 | y | n | n | n
Windows 10 x64 | y | n | n | n
Windows 10 Cygwin | n | n | x | n
Mac OS X El Capitan 10.11.6 64-bit | y | n | y | n
Mac OS Sierra 10.12.6 64-bit | y | n | y | n
Mac OS X High Sierra 10.13.6 64-bit | y | n | y | n
Mac OS X Mojave 10.14.6 64-bit | y | n | y | n
CentOS 7.2 Linux 3.10.0 x86_64 PGI | y | n | y | n
CentOS 7.2 Linux 3.10.0 x86_64 GNU | y | y | y | y
CentOS 7.2 Linux 3.10.0 x86_64 Intel | y | y | y | y
Linux 2.6.32-573.18.1.el6.ppc64 | y | n | y | n

<p>[1] Parallel

**Comment by gsjaardema** [comment]
The links don’t work for me.

**Reply by brtnfld** [reply]
I'm not sure why cgns.org/"page" is not working, cgns.org correctly redirects to the github pages. Strange.
